src/waveform/renderers/waveformbeat.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <proto/beats.pb.h>
+
+#include <QPainter>
+
+using BeatType = mixxx::track::io::Type;
+
+enum Direction {
+    UP = 0,
+    DOWN,
+    LEFT,
+    RIGHT
+};
+
+class WaveformBeat {
```

Comment:
Either declare a virtual destructor or declare this class as `final`. The latter may apply here.